File: word2json/models/casefile.py
# %%
from typing import Optional
from word2json.models.base import *
from datetime import datetime, date


class Casefile(BaseDocument):

    body: Optional[list[Body]]
    category: Optional[str]
    description: Optional[str]
    featured: Optional[bool]
    header: Optional[str]
    mainImageUrl: Optional[str]
    order: Optional[int | float]
    slug: Optional[Slug]
    tags: Optional[list[str]]
    title: Optional[str]
    writtenAt: Optional[datetime]
    publishedAt: Optional[datetime]
    imageUrls: Optional[list[ImageUrl]]
    articleUrls: Optional[list[ArticleUrl]]

File: json2word/models/casefile.ts
import { Body, Slug, BaseDocument, ImageUrl, ArticleUrl } from "./base";

export interface Casefile extends BaseDocument {
	body?: Body[];
	category?: string;
	description?: string;
	featured?: boolean;
	header?: string;
	mainImageUrl?: string;
	order?: number;
	slug: Slug;
	tags?: string[];
	title: string;
	writtenAt: string;
	publishedAt: string;
	imageUrls?: ImageUrl[];
	articleUrls?: ArticleUrl[];
}
